components/script/dom/node.rs

```diff
@@ -346,13 +349,16 @@ impl Node {
             // rely on the state of IS_IN_DOC of the context node's descendants,
             // e.g. when removing a <form>.
             vtable_for(&node).unbind_from_tree(context);
-            // https://dom.spec.whatwg.org/#concept-node-remove step 14
```

Comment:
And I understand that this existing comment would seem to indicate that this is how it should be document, but I assume this is old code that hasn't been revisited in a while, hence the opportunity to improve things while we're at it.